lib/api/apiUtils/bucket/models/ReplicationConfiguration.js

```diff
@@ -92,6 +93,25 @@ class ReplicationConfiguration {
     }
 
     /**
+     * Generate a 48-character alphanumeric, unique ID for a rule
+     * @param {number} [n] - The retry count, if calling recursively
+     * @return {string} - The unique rule ID
+     */
+    _createUniqueID(n) {
+        // If no ID is given in the replication configuration, we create it.
+        const id = Buffer.from(UUID.v4()).toString('base64');
+        // Each ID in a list of rules must be unique. If the generated ID
+        // already exists, call recursively until a unique ID is generated or
+        // we have retried five times.
+        let retries = n || 0;
```

Comment:
They are random. The check was just to prevent the (very) unlikely chance that it already exists. I would be happy to remove the check though, if it is too unlikely and thus not worth it.